agent/src/beerocks/slave/tasks/task.h

```diff
@@ -0,0 +1,70 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2016-2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _TASK_H_
+#define _TASK_H_
+
+#include <beerocks/tlvf/beerocks_header.h>
+#include <tlvf/CmduMessageRx.h>
+
+namespace beerocks {
+
+class Task {
+public:
+    Task() {}
+    virtual ~Task() {}
+
+    /**
+     * @brief Enum which defines all tasks type.
+     * 
+     * Each task must have an entry on the enum eType, so it will be possible to send the task an
+     * event. 
+     */
+    enum eType : uint8_t {
+        TOPOLOGY,
+    };
+
+    /**
+     * @brief Retrun the task type.
+     * 
+     * Called by TaskPool send_event().
+     * 
+     * @return eType Task type.
+     */
+    virtual eType get_task_type() = 0;
+
+    /**
+     * @brief Work function. Shall do the task routine. Called by TaskPool run_task() function.
+     */
+    virtual void work(){};
+
+    /**
+     * @brief Handle events, which are being send from containing thread.
+     * 
+     * @param event_enum_value Event enum value which shall be defined on the task.
```

Comment:
If we expect enum, it is not stated in the interface.
If we expect uint8_t then we shouldn't mention enums. It is up to the user to decide what this uint8_t is representing.

Reply:
It is stated:
"@param event_enum_value **Event enum value which shall be defined on the task**."

Wasn't it clear enough?

Reply:
nitpick, m2c -
Comments are not the interface, they are used as a complementary for the code. They should not contradict it or add embarrassment to the reader.
In this specific case: there is the word `enum` both in the comment and in the variable name. But the _type_ of the variable is `unit8_t`.
So now, as a user I am confused: should I declare an enum? what this enum represent? etc.
The fact that `unit8_t` can be converted to to an enum is not important.
For example I can use this `uint8_t` value as an index to a array, or use it as the code of a CMDU message.
The intent of the function is not clear, it is not aligned with the interface.
It might look like of over-engineering, or overlooked issue - after all, what is the issue here, the comment? - when occurred once it is not an issue, but when a project have more and more of these, they make it harder to maintain, and of course bugs are introduced.

Reply:
I try to explain it as much as I could to make it clear.
I wanted that each task will define its own event, which will be passed through the task_pool.
On the task pool, I must give it a uint8_t.
If you have a suggestion on how to do it more clear, please enlighten me.
Keeping the task object on the caller class without using the task pool is not an option.

Reply:
> Keeping the task object on the caller class without using the task pool is not an option.

This is the best option.
Other than that:
```suggestion
     * @param message uint8_t that represent anything in the context of the targeted Task
```